solutions/2018/02_inventory_management_system.rb

```diff
@@ -0,0 +1,37 @@
+class InventoryManager
+  def initialize(box_ids)
+    @box_ids = box_ids
+  end
+
+  def checksum
+    twos_count = threes_count = 0
+
+    @box_ids.each do |id|
+      twos_count += 1 if has_any_element_frequency?(id, 2)
+      threes_count += 1 if has_any_element_frequency?(id, 3)
+    end
+
+    twos_count * threes_count
+  end
+
+  def close_match
+    @box_ids.combination(2).each do |pair|
```

Comment:
Again, here I would move this to a method on a smaller class, so you could do something like `box_id.close_match?(other_box_id)` or the like, and here you just _use_ that information that the smaller class provided.

Also, if you do `.each do |(first_id, last_id)|` you can destructure the array while you iterate over it